cmake/modules/AddDoxygenDocs.cmake

```
@@ -0,0 +1,144 @@
+macro(add_doxygen_docs)
+    set(dirs
+        apriltag
+        cameraserver
+        cscore
+        fieldImages
+        hal
+        ntcore
+        romiVendordep
+        wpilibc
+        wpilibNewCommands
+        wpimath
+        wpinet
+        wpiutil
+        xrpVendordep
```

Why are ntcoreffi, simulation and wpigui excluded?
```suggestion
        apriltag
        cameraserver
        cscore
        fieldImages
        hal
        ntcore
        ntcoreffi
        romiVendordep
        simulation
        wpigui
        wpilibc
        wpilibNewCommands
        wpimath
        wpinet
        wpiutil
        xrpVendordep
```

---

I see now we don't build them in Gradle but idk why.

---

Gradle for some god unknown reason doesnt work

---

wpigui *should* work to build docs with but doesnt

---

ntcoreffi, simulation, and wpigui are not part of the team-facing API surface. 
ntcoreffi is exports for LabVIEW (networktables, DataLogManager)
The simulation subproject contains the HAL plugins (backends for HALSIM, no public API)
wpigui is only used internally to the GUI apps.